apps/dotcms-ui/src/app/view/components/_common/dot-content-type-clone-dialog/dot-content-type-clone-dialog.component.ts

```diff
@@ -0,0 +1,119 @@
+import {
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    EventEmitter,
+    Input,
+    OnDestroy,
+    OnInit,
+    Output
+} from '@angular/core';
+import { LoggerService } from '@dotcms/dotcms-js';
+import { DotDialogActions } from '@components/dot-dialog/dot-dialog.component';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { DotMessageService } from '@services/dot-message/dot-messages.service';
+import { DotCMSAssetDialogData } from '@dotcms/dotcms-models';
+import { takeUntil } from 'rxjs/operators';
+import { Subject } from 'rxjs';
+
+@Component({
+    selector: 'dot-dot-content-type-clone-dialog',
+    templateUrl: './dot-content-type-clone-dialog.component.html',
+    styleUrls: ['./dot-content-type-clone-dialog.component.css'],
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class DotContentTypeCloneDialogComponent implements OnInit, OnDestroy {
+    dialogActions: DotDialogActions;
+    inputNameWithType = '';
+    @Input()
+    isVisibleDialog = false;
+    @Output() cancel = new EventEmitter<boolean>();
+    form!: FormGroup;
+
+    asset: DotCMSAssetDialogData;
+
+    private destroy$: Subject<boolean> = new Subject<boolean>();
+
+    constructor(
+        private readonly loggerService: LoggerService,
+        private readonly fb: FormBuilder,
+        private readonly dotMessageService: DotMessageService,
+        private readonly cd: ChangeDetectorRef
+    ) {}
+
+    @Input()
+    set assetSelected(assetSelected: DotCMSAssetDialogData | null) {
+        if (assetSelected) {
+            this.asset = assetSelected;
+            this.inputNameWithType = this.getNameFieldLabel(assetSelected.baseType);
+        }
+    }
+
+    ngOnInit(): void {
+        this.initForm();
+        this.setDialogConfig();
+    }
+
+    submitForm() {
+        // to implement
+        //console.info('submitForm: ', this.form.value);
+    }
+
+    closeDialog(): void {
+        this.cancel.emit(true);
+        this.initForm();
+    }
+
+    ngOnDestroy(): void {
+        this.destroy$.next(true);
+        this.destroy$.complete();
+    }
+
+    ngAfterViewChecked() {
```

Comment:
To use this method of Angular Lifecycle, we need to implement [AfterViewChecked](https://angular.io/api/core/AfterViewChecked) in the class.

Reply:
Added